### PR TITLE
Update deps

### PR DESCRIPTION
**Pull request by saitheninja**
Hi. I was browsing the docs and I noticed that maybe I could help clean up some things.
Related issue: #228

Things done in this pull request:
- Update Bootstrap from 5.0 beta to 5.3.3.
- Remove old normalize.css. Bootstrap already includes its own [reboot](https://getbootstrap.com/docs/5.3/content/reboot/).
- Remove jquery. It isn't a dependency of Bootstrap in v5. The bootstrap.bundle.js includes popper.js as a replacement.
- Update nix-shell, because the ruby version was too old to build.
- Update navbar to match previous styling. There's also some new CSS classes to target things better.
- Update fonts. Serif fonts weren't being used at all, and Inter wasn't being loaded in some places. Also Lato isn't being used at all anymore, so I removed the old files.
- Clean up some CSS.

Everything built fine on my machine. The only visual change is using the Inter font.
Screenshot_20240520_173925

Also there's some scripts that need to get their references updated to the correct filenames:
on the main site:
https://github.com/neovim/neovim/blob/40ce8577977fcdce8ad76863c70eb522e4cefd4d/scripts/gen_help_html.lua#L829
And the docs:
https://github.com/neovim/doc/blob/6f805e0d6085bd29ef881dcfe907638c943aac75/templates/report-header.sh.html#L11

I can do pull requests to update those if everything looks good, I just wanted to make sure that I didn't miss anything, or do something dumb. This pull request got a little bigger than I expected.

### CODE REVIEW

**Comment by justinmk** [comment]
This is definitely wanted, but why are there 18k lines added?

**Comment by saitheninja** [comment]
Good question. I think that's mostly the unminified versions of bootstrap.css and bootstrap.bundle.js.

I included the unminified versions because it's easier to browse through without having to load up a dev environment. Also the repo currently has the unminified and minified versions so I was just pattern matching. But I dont have any strong feelings about including it.

Should I go ahead and remove the unminified files?

**Comment by saitheninja** [comment]
Sure. Give me like 30 minutes and I'll clean it up.

**Comment by saitheninja** [comment]
Okay, cleaned it up. The diff is now much more reasonable.

Also made pull requests to update the links from the other repos to point to the correct files.

File: css/main.css [hunk 1]
```diff
@@ -49,15 +43,12 @@
...
-      var(--bs-font-sans-serif);
-    --serif-fonts: "Inter var", var(--serif-fonts), var(--bs-font-serif);
+    --sans-serif-fonts: "Inter var", var(--bs-font-sans-serif);
   }
 }
 
 body {
-  /* font-family: "Helvetica Neue", Helvetica, Calibri, Arial, sans-serif; */
-  font-family: var(--serif-fonts);
+  font-family: var(--sans-serif-fonts);
```
Comment (justinmk) [inline css/main.css:51]: We definitely weren't using serif, I wonder why this was like this.

**Review by justinmk — APPROVED** [review]
This is extremely helpful. Keeping up with details like "normalize.css vs reboot" takes attention. Thank you!

**Pull request merged**